repository/src/main/java/org/alfresco/repo/action/executer/NodeSizeDetailActionExecutor.java

```diff
@@ -0,0 +1,213 @@
+            response.put(EXCEPTION,"Exception occurred while parsing String to INT: {} " + numberFormatException.getMessage());
+            simpleCache.put(actionedUponNodeRef.getId(),response);
+            throw numberFormatException;
+        }
+
+        NodeRef nodeRef = actionedUponNodeRef;
+        long totalSizeFromFacet = 0;
+        int skipCount = 0;
+        ResultSet results;
+        int totalItems;
+        boolean isCalculationCompleted = false;
+
+        try
+        {
+            // executing Alfresco FTS facet query.
+            results = facetQuery(nodeRef);
+            totalItems = Math.min(results.getFieldFacet(FIELD_FACET).size(), defaultItems);
+            // Using AtomicLong to accumulate the total size.
+            AtomicLong resultSize = new AtomicLong(0);
+
+            while (!isCalculationCompleted)
+            {
+                List<Pair<String, Integer>> pairSizes = results.getFieldFacet(FIELD_FACET).subList(skipCount, totalItems);
+                pairSizes.parallelStream().forEach(id -> {
+                    try
+                    {
+                        if(id.getSecond()>0)
+                        {
+                            resultSize.addAndGet(Long.valueOf(id.getFirst()) * id.getSecond());
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        resultSize.addAndGet(0);
+                    }
+                });
+
+                totalSizeFromFacet+=resultSize.longValue();
+                resultSize.set(0);
+
+                if (results.getFieldFacet(FIELD_FACET).size() <= totalItems || results.getFieldFacet(FIELD_FACET).size() <= defaultItems)
+                {
+                    isCalculationCompleted = true;
+                }
+                else
+                {
+                    skipCount += defaultItems;
+                    int remainingItems = results.getFieldFacet(FIELD_FACET).size() - totalItems;
+                    totalItems += Math.min(remainingItems, defaultItems);
+                }
+            }
+        }
+        catch (RuntimeException runtimeException)
+        {
+            LOG.error("Exception occurred in NodeSizeDetailActionExecutor:results {} ", runtimeException.getMessage());
+            response.put(EXCEPTION,"Exception occurred in NodeSizeDetailActionExecutor:results {} "+runtimeException.getMessage());
+            simpleCache.put(nodeRef.getId(),response);
+            throw runtimeException;
+        }
+
+        LOG.debug(" Calculating size of Folder Node - NodeSizeDetailActionExecutor:executeImpl ");
+        final LocalDateTime eventTimestamp = LocalDateTime.ofInstant(Instant.now(), ZoneId.systemDefault());
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy:MM:dd HH:mm:ss");
+        String formattedTimestamp = eventTimestamp.format(formatter);
```

Comment:
There's no need to format the date here, as the format used does not comply with our REST API guidelines. I noticed you are using this value as-is in the API response. The REST framework will automatically convert the date object to ISO8601 format, so you can simplify the code by removing the manual formatting."

Reply:
Done